include/pybind11/cast.h

```diff
@@ -322,7 +322,7 @@ template <> class type_caster<bool> {
             }
             #endif
             if (res == 0 || res == 1) {
-                value = (bool) res;
```

Comment:
Why not just make these `value = res != 0`; ? The problem is that (bool) int_value is poor for performance (according to the warning), so we shouldn't be pretending this is okay to cast.

Reply:
Done. Applied globally. — I was aiming for a robust general solution before, but you're right, we only need specific solutions, and `!= 0` is simpler therefore better.